### PR TITLE
Use UriHelper.BuildAbsolute and UriHelper.BuildRelative for RedirectRule

### PR DESCRIPTION
**Pull request by paulomorgado**
Use `UriHelper.BuildAbsolute` and `UriHelper.BuildRelative` for `RedirectRule` URL generation.

Addresses #28883

**NOTE;:** The current implementation does not correctly encode all URLs.

### CODE REVIEW

**Comment by jkotalik** [comment]
> The current implementation does not correctly encode all URLs.

What do you mean by this? I may have missing context here.

**Comment by paulomorgado** [comment]
If you run the added unit tests to the current implementation, you'll see,

Some of the added tests fail in the current version.

**Comment by jkotalik** [comment]
I currently don't have cycles to finish this PR, @paulomorgado can you finish this PR and or close if you aren't able to figure out how to fix the encoding of URLs.

**Review by jkotalik — APPROVED** [review]

File: src/Middleware/Rewrite/src/RedirectRule.cs [hunk 1]
```diff
@@ -27,60 +28,85 @@ public RedirectRule(string regex, string replacement, int statusCode)
                 throw new ArgumentException(nameof(replacement));
             }
 
-            InitialMatch = new Regex(regex, RegexOptions.Compiled | RegexOptions.CultureInvariant, _regexTimeout);
-            Replacement = replacement;
-            StatusCode = statusCode;
+            this.InitialMatch = new Regex(regex, RegexOptions.Compiled | RegexOptions.CultureInvariant, this._regexTimeout);
```
Comment (jkotalik) [inline src/Middleware/Rewrite/src/RedirectRule.cs:31]: nit: remove `this`

**Pull request merged**